.github/workflows/tests.yml

```diff
@@ -23,6 +23,7 @@ jobs:
           - {python-version: 3.7, interfaces: ['tf']}
           - {python-version: 3.8, interfaces: ['tf', 'torch', 'jax']}
           - {python-version: 3.9, interfaces: []}
+          - {python-version: 3.9, interfaces: ['numba']}

```

Comment:
Let's instead test the numba device as a device, see below lines 86+.  This means the device-specific tests won't be run, but this way we will get coverage of the device in a more contained way. 

If we add a new entry here, all the test suite will run again, even though we just want to test a tiny section of code.  We could potentially add `'numba'` to one of the existing sets here instead of creating a whole new set, but numba is restrictive on the numpy version.  We want to test our suite with the latest numpy version, not an outdated one that works for numba.

Taking that into consideration, I think our best bet is to add it to the device-test suite.